espnet/nets/pytorch_backend/decoders.py

```diff
@@ -75,19 +75,21 @@ def __init__(self, eprojs, odim, dlayers, dunits, sos, eos, att, verbose=0,
     def zero_state(self, hs_pad):
         return hs_pad.new_zeros(hs_pad.size(0), self.dunits)
 
-    def forward(self, hs_pad, hlens, ys_pad):
+    def forward(self, hs_pad, hlens, ys_pad, strm_idx=0):
         """Decoder forward
 
         :param torch.Tensor hs_pad: batch of padded hidden state sequences (B, Tmax, D)
         :param torch.Tensor hlens: batch of lengths of hidden state sequences (B)
         :param torch.Tensor ys_pad: batch of padded character id sequence tensor (B, Lmax)
+        :param int strm_idx: stream index for attention ModuleList when args.spa (speaker parallel attention) is True
         :return: attention loss value
         :rtype: torch.Tensor
         :return: accuracy
         :rtype: float
         """
         # TODO(kan-bayashi): need to make more smart way
         ys = [y[y != self.ignore_id] for y in ys_pad]  # parse padded ys
+        att_idx = min(strm_idx, len(self.att) - 1)
```

Comment:
Why do you use different name? (`strm_idx` and `att_idx`)

Reply:
Stream index (strm_idx) indicates the index of decoding stream. Attention index (att_idx) indicates the index of the attention module. In Non-SPA multi-speaker case, att_idx is always 0. However, the strm_idx may be 0 or 1.

Reply:
@simpleoier, can you add some comments on the source code about the strm_idx and att_idx?

Reply:
OK. Already done.